packages/ai/core/generate-text/stream-text.ts

```diff
@@ -1591,10 +1592,11 @@ However, the LLM results are expected to be small enough to not cause issues.
     );
   }
 
-  async consumeStream(): Promise<void> {
-    const stream = this.fullStream;
-    for await (const part of stream) {
-      // no op
+  async consumeStream(onError?: (error: unknown) => void): Promise<void> {
+    try {
+      await consumeStream(this.fullStream);
```

Comment:
wonder if it's worth pushing the onError object into `consumeStream` to keep the behavior between `consumeStream` here and the general helper aligned?

Reply:
the helper is afaik only used internally so we can change the main `stream` object to become a parameter object 

```
{
  stream,
  onError?
}
```